sympy/geometry/ellipse.py

```diff
@@ -695,25 +693,50 @@ def is_tangent(self, o):
         True
 
         """
-        inter = None
-        if isinstance(o, Ellipse):
-            inter = self.intersection(o)
-            if isinstance(inter, Ellipse):
+        if isinstance(o, Point2D):
+            return False
+        elif isinstance(o, Ellipse):
+            intersect = self.intersection(o)
+            if isinstance(intersect, Ellipse):
+                return True
+            elif intersect:
+                if all([(self.tangent_lines(i)[0]).equals((o.tangent_lines(i)[0])) for i in intersect]):
+                    return True
+                else:
+                    return False
+            else:
+                return False
```

Comment:
This can be cleaned up.

```
            intersect = self.intersection(o)
            if isinstance(intersect, Ellipse):
                return True
            elif intersect:
                if all([(self.tangent_lines(i)[0]).equals((o.tangent_lines(i)[0])) for i in intersect]):
                    return True
            return False
```